drivers/docker/driver.go

```diff
@@ -0,0 +1,1682 @@
+		"cap_add":        hclspec.NewAttr("cap_add", "list(string)", false),
+		"cap_drop":       hclspec.NewAttr("cap_drop", "list(string)", false),
+		"command":        hclspec.NewAttr("command", "string", false),
+		"cpu_hard_limit": hclspec.NewAttr("cpu_hard_limit", "bool", false),
+		"cpu_cfs_period": hclspec.NewAttr("cpu_cfs_period", "number", false),
+		"devices": hclspec.NewBlockSet("devices", hclspec.NewObject(map[string]*hclspec.Spec{
+			"host_path":          hclspec.NewAttr("host_path", "string", false),
+			"container_path":     hclspec.NewAttr("container_path", "string", false),
+			"cgroup_permissions": hclspec.NewAttr("cgroup_permissions", "string", false),
+		})),
+		"dns_search_domains": hclspec.NewAttr("dns_search_domains", "list(string)", false),
+		"dns_options":        hclspec.NewAttr("dns_options", "list(string)", false),
+		"dns_servers":        hclspec.NewAttr("dns_servers", "list(string)", false),
+		"entrypoint":         hclspec.NewAttr("entrypoint", "list(string)", false),
+		"extra_hosts":        hclspec.NewAttr("extra_hosts", "list(string)", false),
+		"force_pull":         hclspec.NewAttr("force_pull", "bool", false),
+		"hostname":           hclspec.NewAttr("hostname", "string", false),
+		"interactive":        hclspec.NewAttr("interactive", "bool", false),
+		"ipc_mode":           hclspec.NewAttr("ipc_mode", "string", false),
+		"ipv4_address":       hclspec.NewAttr("ipv4_address", "string", false),
+		"ipv6_address":       hclspec.NewAttr("ipv6_address", "string", false),
+		"labels":             hclspec.NewAttr("labels", "map(string)", false),
+		"load":               hclspec.NewAttr("load", "string", false),
+		"logging":            hclspec.NewAttr("logging", "map(string)", false),
+		"mac_address":        hclspec.NewAttr("mac_address", "map(string)", false),
+		"mounts": hclspec.NewBlockSet("mounts", hclspec.NewObject(map[string]*hclspec.Spec{
+			"target":   hclspec.NewAttr("target", "string", false),
+			"source":   hclspec.NewAttr("source", "string", false),
+			"readonly": hclspec.NewAttr("readonly", "bool", false),
+			"volume_options": hclspec.NewBlockSet("volume_options", hclspec.NewObject(map[string]*hclspec.Spec{
+				"no_copy": hclspec.NewAttr("no_copy", "bool", false),
+				"labels":  hclspec.NewAttr("labels", "map(string)", false),
+				"driver_config": hclspec.NewBlockSet("driver_config", hclspec.NewObject(map[string]*hclspec.Spec{
+					"name":    hclspec.NewAttr("name", "string", false),
+					"options": hclspec.NewAttr("name", "map(string)", false),
+				})),
+			})),
+		})),
+		"network_aliases": hclspec.NewAttr("network_aliases", "list(string)", false),
+		"network_mode":    hclspec.NewAttr("network_mode", "string", false),
+		"pids_limit":      hclspec.NewAttr("pids_limit", "number", false),
+		"pid_mode":        hclspec.NewAttr("pid_mode", "string", false),
+		"port_map":        hclspec.NewAttr("port_map", "map(number)", false),
```

Comment:
All of the fields that were `[]map[string]XXX` (https://github.com/hashicorp/nomad/blob/40a5914c87138f5efbd45a23b162fbc75dac8743/client/driver/docker.go#L213) should become `NewBlockAttrs`: https://github.com/hashicorp/nomad/blob/master/drivers/rkt/driver.go#L114